bats/foreign-keys.bats

```diff
@@ -1511,3 +1510,277 @@ SQL
   [ "$status" -eq "0" ]
   [[ "$output" =~ "Warning" ]] || false
 }
+
+@test "foreign-keys: self-referential same column(s)" {
+    dolt sql <<SQL
+CREATE INDEX v1v2 ON parent(v1, v2);
+SQL
+    run dolt sql -q "ALTER TABLE parent ADD CONSTRAINT fk_named FOREIGN KEY (v1) REFERENCES parent(v1);"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ 'the same column `v1` cannot be used in self referential foreign keys' ]] || false
+    run dolt sql -q "ALTER TABLE parent ADD CONSTRAINT fk_named FOREIGN KEY (v1, v2) REFERENCES parent(v1, v2);"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "self referential" ]] || false
+}
+
+@test "foreign-keys: self-referential child column follows parent RESTRICT" {
+    # default reference option is RESTRICT
+    dolt sql <<SQL
+ALTER TABLE parent ADD CONSTRAINT fk_named FOREIGN KEY (v2) REFERENCES parent(v1);
+INSERT INTO parent VALUES (1,1,1), (2, 2, 1), (3, 3, NULL);
+UPDATE parent SET v1 = 1 WHERE id = 1;
+UPDATE parent SET v1 = 4 WHERE id = 3;
+DELETE FROM parent WHERE id = 3;
+SQL
+    run dolt sql -q "SELECT * FROM parent" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "id,v1,v2" ]] || false
+    [[ "$output" =~ "1,1,1" ]] || false
+    [[ "$output" =~ "2,2,1" ]] || false
+    [[ "${#lines[@]}" = "3" ]] || false
+
+    run dolt sql -q "DELETE FROM parent WHERE v1 = 1;"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "violation" ]] || false
+    run dolt sql -q "UPDATE parent SET v1 = 2;"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "violation" ]] || false
+    run dolt sql -q "REPLACE INTO parent VALUES (1, 1, 1);"
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "violation" ]] || false
+}
+
+@test "foreign-keys: self-referential child column precedes parent RESTRICT (SKIPPED)" {
+    # Once we implement this functionality then we need to delete this statement to let the rest of the test run, and edit the title.
+    # This ensures that the test is not ignored, unlike with a skipped test.

```

Comment:
The idea was that these would fail if we ever implemented the logic but didn't "_unskip_" the test, so this makes it an explicit failure. We have tests that are/were skipped and they now work, and in some cases the tests are flat out wrong OR the logic implemented doesn't work according to the test because the implementor either didn't know that the test existed or forgot about it.

With this error and return 0, the PR will fail if the logic is touched and the implementor doesn't change it, which I feel is a FAR superior solution to skipped tests. Skipped tests make them optional, and assuming the tests are correct, I don't like optional things.

It's a non issue though cause I'mma rip them out anyway.